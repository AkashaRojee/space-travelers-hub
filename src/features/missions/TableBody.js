import propTypes from 'prop-types';

const TableBody = ({ missions, onJoinMission }) => (
  <tbody>
    {
      missions.map(({
        mission_id: id, mission_name: name, description, joined,
      }) => (
        <tr key={id}>
          <td>{name}</td>
          <td>
            {description}
          </td>
          <td>
            <span className={`${joined ? 'joined' : ''}`}>{!joined ? 'NOT A MEMBER' : 'Active Member' }</span>
          </td>
          <td>
            <button type="button" className={`${joined ? 'joined' : ''}`} onClick={() => onJoinMission(id, joined)}>{!joined ? 'Join Mission' : 'Leave Mission' }</button>
          </td>
        </tr>
      ))
    }
  </tbody>
);

TableBody.propTypes = {
  missions: propTypes.arrayOf(propTypes.any).isRequired,
  onJoinMission: propTypes.func.isRequired,
};

export default TableBody;
