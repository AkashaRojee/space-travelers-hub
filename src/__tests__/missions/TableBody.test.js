import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TableBody from '../../features/missions/TableBody';

const missions = [
  {
    mission_name: 'Thaicom',
    mission_id: '9D1B7E0',
    description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
  },
];

const join = (id) => id;

describe('Table body component', () => {
  beforeEach(() => {
    render(<TableBody missions={missions} onJoinMission={join} />);
  });
  it('should match the snapshot', () => {
    const tree = renderer.create(<TableBody missions={missions} onJoinMission={join} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change to leave mission', () => {
    fireEvent.click(screen.getByText('Join Mission'));
    const result = screen.getByTitle('badge');
    expect(result.memoizedProps).not.toBe('NOT A MEMBER');
  });
});
