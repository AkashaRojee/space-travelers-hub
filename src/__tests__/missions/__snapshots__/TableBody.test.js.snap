// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table body component should match the snapshot 1`] = `
<tbody>
  <tr>
    <td>
      Thaicom
    </td>
    <td>
      Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
    </td>
    <td>
      <span
        className=""
        title="badge"
      >
        NOT A MEMBER
      </span>
    </td>
    <td>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>
</tbody>
`;
